src/validation.cpp

```diff
@@ -4501,7 +4512,10 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
         if (nCheckLevel >= 3 && (coins.DynamicMemoryUsage() + ::ChainstateActive().CoinsTip().DynamicMemoryUsage()) <= nCoinCacheUsage) {
             assert(coins.GetBestBlock() == pindex->GetBlockHash());
             if (pindex->nHeight >= chainparams.GetConsensus().nStakingStartHeight) {
-                assert(stakes.getBestBlock() == pindex->GetBlockHash());
+                if (stakes.getBestBlock() != pindex->GetBlockHash()) {
+                    return error("Found inconsistency in stakes DB. Restart with --reindex-chainstate flag to rebuild it.");
+                }
```

Comment:
Should be == or !=?
Assert was with ==, but now its different. Just checking

Reply:
assert was supposed to throw on condition fail, now there is an error thrown on condition fulfilled.